src/main/java/org/htmlunit/util/StringUtils.java

```diff
@@ -49,8 +49,8 @@ public final class StringUtils {
                                 + "\\s*((0|[1-9]\\d?|100)(.\\d*)?)%\\s*\\)");
     private static final Pattern ILLEGAL_FILE_NAME_CHARS = Pattern.compile("\\\\|/|\\||:|\\?|\\*|\"|<|>|\\p{Cntrl}");
 
-    private static final Map<String, String> CamelizeCache_ = new HashMap<>();
-    private static final Map<String, String> RootLowercaseCache_ = new HashMap<>();
+    private static final Map<String, String> CamelizeCache_ = new ConcurrentHashMap<>();
+    private static final Map<String, String> RootLowercaseCache_ = new ConcurrentHashMap<>();
```

Comment:
RootLowercaseCache_  seems to be no longer used - does having this case makes no sense?

Reply:
found the answer in the commit comment